Support adding a note + attachment in one step

vboctor opened the pull request:
Fixes #19589

atrol commented on bugnote_add_inc.php:
```diff
@@ -127,6 +127,34 @@
+				<th class="category">
+					<?php echo lang_get( $t_file_upload_max_num == 1 ? 'upload_file' : 'upload_files' ) ?>
+					<br />
+					<?php echo print_max_filesize( $t_max_file_size ); ?>
```
remove echo, print_max_filesize is void and does the echo itself
I am aware you copied the code from bug_file_upload_inc.php

dregad replied:
If the code is indeed duplicated, then I think it should be moved into an API function instead.

vboctor commented:
I've done some re-factoring to drive consistency for the following:
- Detecting when files are not provided (before we were using null vs. -1).
- Checking for access level, now we always use method to check if user can upload rather than generic access check.
- Added a new method in file_api to process files.
- Removed the echo of the print method (was in the bug uploads, but not project doc uploads).

As for the duplication of the UI logic, I haven't refactored this for the following reasons:
- Modern UI has this changed.
- This code was already duplicated on bug report page, bug view page, and upload for project files.
- Supporting drag and drop attachments will change the html code.
- Labels are not exactly the same, sometime we use "select files" under title "upload files".  And sometimes when there is no time, we just use "upload file".
- The UI code is not exactly the same.
